packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -13,6 +13,7 @@ import WarningFilled16 from '@carbon/icons-react/lib/warning--filled/16';
 import CaretDownGlyph from '@carbon/icons-react/lib/caret--down/index';
 import CaretUpGlyph from '@carbon/icons-react/lib/caret--up/index';
 import mergeRefs from '../../tools/mergeRefs';
+import { useControlledStateWithEventListener } from '../../internal/FeatureFlags';
```

Comment:
Flipping the switch upon React context is in consideration, for the time once we have finished implementing this mechanism to all form components.